rules/preprocessing_methyldackel.py

```diff
@@ -0,0 +1,199 @@
+# PiGx BSseq Pipeline.
+#
+# Copyright 2019 Alexander Blume <alexander.blume@mdc-berlin.de>
+#
+# This file is part of the PiGx BSseq Pipeline.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# This file was originally taken and modified from
+# https://github.com/katwre/makeWGBSnake/blob/master/Rules/Meth_preprocessing_methyldackel_rules.py
+
+# WGBS pipeline
+#
+# Copyright © 2018 Katarzyna Wreczycka katarzyna.wreczycka@mdc-berlin.de
+# This pipeline is heavily based on the PiGx BSseq pipeline github.com/BIMSBbioinfo/pigx_bsseq
+
```

Comment:
I guess in this case (Kasia copies from bsseq, and you copy from Kasia's copy) you can just add Kasia's copyright line to the top and remove the provenance comments.